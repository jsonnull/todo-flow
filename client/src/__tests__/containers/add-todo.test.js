import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import reducer from '../../reducers';
import AddTodo from '../../containers/add-todo';

describe('AddTodo', () => {
  let tree;
  let store;
  beforeEach(() => {
    store = createStore(reducer);
  });
  it('renders without crashing', () => {
    const component = renderer.create(
      <Provider store={store}><AddTodo /></Provider>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('onSubmit', () => {
    let component;
    beforeEach(() => {
      component = renderer.create(
        <Provider store={store}><AddTodo /></Provider>
      );
      tree = component.toJSON();
    });

    xit('calls preventDefault to stop form submit');

    describe('with no input value', () => {
      xit('returns early');
    });
  });
});
