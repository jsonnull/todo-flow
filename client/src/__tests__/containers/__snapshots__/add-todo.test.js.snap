exports[`AddTodo renders without crashing 1`] = `
<div>
  <form
    onSubmit={[Function]}>
    <input />
    <button
      type="submit">
      Add Todo
    </button>
  </form>
</div>
`;
